pkg/controller/machine/machine_controller.go

```diff
@@ -721,6 +724,27 @@ func (r *Reconciler) ensureInstanceExistsForMachine(
 				externalCloudProvider, _ = strconv.ParseBool(val)
 			}
 
+			registryCredentials := map[string]containerruntime.AuthConfig{}
+
+			if secRef := strings.SplitN(r.nodeSettings.RegistryCredentialsSecretRef, "/", 2); len(secRef) == 2 {
+				var credsSecret corev1.Secret
+				err := r.client.Get(ctx, types.NamespacedName{Namespace: secRef[0], Name: secRef[1]}, &credsSecret)
+				if err != nil {
+					return nil, fmt.Errorf("failed to retrieve registry credentials secret object: %w", err)
+				}
+
+				for registry, data := range credsSecret.Data {
+					var regCred containerruntime.AuthConfig
+					if err := json.Unmarshal(data, &regCred); err != nil {
+						return nil, fmt.Errorf("failed to unmarshal registry credentials: %w", err)
+					}
+					registryCredentials[registry] = regCred
+				}
+			}
```

Comment:
Would be nice to have an error or at the very least a warning `if len(secRef) != 2` because at the moment it would just get skipped past

Reply:
changed validation of the value on the fatal error on the start up